cmd/nodepool/core/create.go

```diff
@@ -142,3 +144,29 @@ func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts
 	fmt.Printf("NodePool %s created\n", o.Name)
 	return nil
 }
+
+// ValidateMgmtClusterSupportsNodePoolCPUType validates the HostedCluster payload type can support the CPU architecture
+// of the NodePool.
+func ValidateMgmtClusterSupportsNodePoolCPUType(ctx context.Context, name, namespace, arch string) error {
```

Comment:
we have
ValidateMgmtClusterSupportsNodePoolCPUType and validateMgmtClusterAndNodePoolCPUArchitectures
Then we have DoesMgmtClusterAndNodePoolCPUArchMatch

the latter is doing a plain comparison, so it could be just called archMatch or similar or is it even necessary?

The others two are confusing to me and seem unconsistent.

Reply:
also ValidateMgmtClusterSupportsNodePoolCPUType needs a unit too

Reply:
Why are we using a new term "cpu type" when we are using "arch" everywhere else?
This PR is using CPUArch cpuType and arch in multiple places to refer to the same thing, can we please just use arch consistently which also matches the api fields naming?

Reply:
Updated